web/ui/react-app/src/pages/flags/Flags.tsx

```diff
@@ -102,29 +102,24 @@ export const FlagsContent: FC<FlagsProps> = ({ data = {} }) => {
           </tr>
         </thead>
         <tbody>
-          {Object.entries(data)
-            .sort(compareAlphaFn(sortState.name === 'Flag', !sortState.alpha))
-            .map(([flag, value]) => [`--${flag}`, value])
-            .filter(([flag, value]) => flag.includes(searchState) || value.includes(searchState))
-            .map(([flag, value]) => {
-              const flagMatchStr = boldSubstring(flag, searchState);
-              const valueMatchStr = boldSubstring(value, searchState);
-              const sanitizeOpts = { allowedTags: ['strong'] };
-              return (
-                <tr key={flag}>
-                  <td className="px-4">
-                    <code className="text-dark flag-item">
-                      <span dangerouslySetInnerHTML={{ __html: sanitizeHTML(flagMatchStr, sanitizeOpts) }} />
-                    </code>
-                  </td>
-                  <td className="px-4">
-                    <code className="text-dark value-item">
-                      <span dangerouslySetInnerHTML={{ __html: sanitizeHTML(valueMatchStr, sanitizeOpts) }} />
-                    </code>
-                  </td>
-                </tr>
-              );
-            })}
+          {filtered.map((result: FilterResult<string>) => {
+            const [flagMatchStr, valueMatchStr] = result.string.split(' ');
```

Comment:
Using the space character as a separator isn't safe unfortunately. If I set `--web.page-title="foo bar"`, then it shows the flag value as simply "foo" in the UI, cutting off anything after the first space in the flag value. I'm not sure there would be any 100% safe characters, but if you chose one that cannot happen in a flag *name* at least (the space should actually be fine for that), then you could still split on that special character, but only on its first occurrence (whether that's by pasting together the rest after the split or by manually finding the index of the first space...). It still seems a bit hacky, but hey...

Reply:
here I choose this separator `||`